tests/manage/z_cluster/nodes/test_check_pod_status_after_two_nodes_shutdown_recovery.py

```diff
@@ -0,0 +1,162 @@
+        )
+        pods_running_nodes = []
+        for pod in pod_obj_list:
+            pods_running_nodes.append(
+                get_node_running_pod(
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=pod.name
+                )
+            )
+
+        log.info(f"{selector} running on nodes {pods_running_nodes}")
+        return pods_running_nodes
+
+    @pytest.mark.polarion_id("OCS-1278")
+    @pytest.mark.polarion_id("OCS-2315")
+    def test_node_after_shutdown_and_recovery_worker_node(self, nodes):
+        """
+        Test case to check MDS pods and Provisioner pods not running on same node
+        post shutdown and recovery node
+
+        """
+
+        # Get MDS pods running nodes before shutdown
+
+        self.list_of_nodes_running_pods(
+            selector='rook-ceph-mds'
+        )
+
+        self.list_of_nodes_running_pods(
+            selector='csi-rbdplugin-provisioner'
+        )
+
+        self.list_of_nodes_running_pods(
+            selector='csi-cephfsplugin-provisioner'
+        )
+
+        # Get the node list
+        node = get_typed_nodes(node_type='worker', num_of_nodes=2)
+
+        # Shutdown 2 worker nodes for 10 mins
+        nodes.stop_nodes(nodes=node)
+
+        waiting_time = 600
+        log.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
+
+        nodes.start_nodes(nodes=node)
+
+        # Validate all nodes are in READY state and up
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=30,
+            delay=15)(
+            wait_for_nodes_status(timeout=1800)
+        )
+
+        # Check the node are Ready state and check cluster is health ok
+        self.sanity_helpers.health_check()
+
+        # Get MDS pods running nodes after recovery
+
+        mds_running_nodes_after_recovery = self.list_of_nodes_running_pods(
+            selector='rook-ceph-mds'
+        )
+
+        rbd_provisioner_running_nodes_after_recovery = self.list_of_nodes_running_pods(
+            selector='csi-rbdplugin-provisioner'
+        )
+
+        cephfs_provisioner_running_nodes_after_recovery = self.list_of_nodes_running_pods(
+            selector='csi-cephfsplugin-provisioner'
+        )
+
+        # Check cephfs plugin provisioner pods running nodes after recovery
+
+        assert len(set(mds_running_nodes_after_recovery)) == len(
+            mds_running_nodes_after_recovery
+        ), "MDS running on same node, Not expected!!!"
+
+        assert len(set(rbd_provisioner_running_nodes_after_recovery)) == len(
+            rbd_provisioner_running_nodes_after_recovery
+        ), "rbd plugin provisioner pods running on Same node, Not expected"
+
+        assert len(set(cephfs_provisioner_running_nodes_after_recovery)) == len(
+            cephfs_provisioner_running_nodes_after_recovery
+        ), "cephfs plugin provisioner pods running on Same node, Not expected"
```

Comment:
You are checking with only running nodes after recovery and not the diff between before and after.
Am I missing anything?

Reply:
After shutdown i check if the list contains same item [compute-0, compute-0], if so the testcase fails
The elements in the list should be different
mds should be running on [compute-0, compute-1] like that